packages/dataviews/src/filter-and-sort-data-view.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import removeAccents from 'remove-accents';
+
+/**
+ * Internal dependencies
+ */
+import {
+	OPERATOR_IS,
+	OPERATOR_IS_NOT,
+	OPERATOR_IS_NONE,
+	OPERATOR_IS_ANY,
+	OPERATOR_IS_ALL,
+	OPERATOR_IS_NOT_ALL,
+} from './constants';
+import { normalizeFields } from './normalize-fields';
+
+function normalizeSearchInput( input = '' ) {
+	return removeAccents( input.trim().toLowerCase() );
+}
+
+const EMPTY_ARRAY = [];
+
+/**
+ * Applies the filtering, sorting and pagination to the raw data based on the view configuration.
+ *
+ * @param {any[]}    data   Raw data.
+ * @param {Object}   view   View config.
+ * @param {Object[]} fields Fields config.
+ *
+ * @return {Object} { data: any[], paginationInfo: { totalItems: number, totalPages: number } }
+ */
+export function filterAndSortDataView( data, view, fields ) {
```

Comment:
I'm not super fan of this name. It misses pagination and uses the redundant dataview suffix.

This utility aims to implement what a `SQL` statement does: filter, sorting, and pagination. It windows data. Alternative names:

- `select( data, view, fields )`
- `prepare( data, view, fields )`

Reply:
I'm not a big fan of the name either. I considered `processDataview` or something like that as well.
I'm not sure I like one word functions but I wouldn't mind either.

Reply:
If we want to go with the existing, at a minimum, I'd rename it to `filterSortAndPaginate` to better communicate what it does (and without the dataviews suffix).